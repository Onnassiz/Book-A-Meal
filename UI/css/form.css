.box {
	width: 400px;
	margin-top: 10px;
	min-height: 300px;
	background: #D7D7D7;
	box-sizing: border-box;
	box-shadow: 0 15px 25px rgba(0,0,0,.5);
	border-radius: 10px;
}
.box #body {
	padding: 20px
}

.box h3{
	color: #4D4D4D;
	font-family: "Trebuchet MS", Helvetica, sans-serif;
	background: #cccecd;
	text-align: center;
	padding: 13px;
	border-top-left-radius: 4px;
	border-top-right-radius: 4px;
}

.box .inputBox {
	position: relative;
}

.box .inputBox input{
	width: 93%;
	padding: 10px 10px 12px 10px;
	font-size: 16px;
	color: #474747;
	background: #dbdfdf;
	border: solid #7E9C18 1px;
	outline: none;
	margin-bottom: 10px;
	margin-top: 8px;
	border-radius: 4px;
}
.box .inputBox textarea{
	width: 93%;
	height: 40px;
	padding: 10px 10px 12px 10px;
	font-size: 16px;
	color: #474747;
	background: #dbdfdf;
	border: solid #7E9C18 1px;
	outline: none;
	margin-bottom: 10px;
	margin-top: 8px;
	border-radius: 4px;
}