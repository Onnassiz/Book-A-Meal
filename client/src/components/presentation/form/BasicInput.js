import React from 'react';
import PropTypes from 'prop-types';

export const BasicInput = props => (
	<div className="inputBox">
		<label htmlFor={props.name}>{props.label}</label>
		<input autoComplete={props.name} className={props.hasError ? 'has-error' : ''} onChange={props.onChange} value={props.value} type={props.type} name={props.name} id={props.name} />
	</div>
);

export const TextArea = props => (
	<div className="inputBox">
		<label htmlFor={props.name}>{props.label}</label>
		<textarea className={props.hasError ? 'has-error' : ''} name={props.name} id={props.name} cols="30" rows="6" onChange={props.onChange} value={props.value} />
	</div>
);

export const Checkbox = props => (
	<label className="checkbox"> {props.meal.name} <span style={{ float: 'right', color: '#7a604a', fontSize: 10 }}>Added by {props.meal.caterer}</span>
		<input type="checkbox" value={props.meal.id} onChange={props.onChange} checked={props.isChecked} /><span className="check" />
	</label>
);

BasicInput.propTypes = {
	name: PropTypes.string.isRequired,
	label: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	onChange: PropTypes.func.isRequired,
	hasError: PropTypes.bool,
};

BasicInput.defaultProps = {
	hasError: false,
};

TextArea.propTypes = {
	name: PropTypes.string.isRequired,
	label: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	onChange: PropTypes.func.isRequired,
	hasError: PropTypes.bool,
};

Checkbox.propTypes = {
	meal: PropTypes.object.isRequired,
	isChecked: PropTypes.bool.isRequired,
	onChange: PropTypes.func.isRequired,
};

TextArea.defaultProps = {
	hasError: false,
};

