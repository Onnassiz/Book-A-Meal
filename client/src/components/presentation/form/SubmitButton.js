import React from 'react';
import PropTypes from 'prop-types';
import loaderImage from '../../../../assets/images/loader.gif';

const SubmitButton = props => (
	<div id="submit">
		<input type="submit" value={props.value} disabled={props.isLoading} />
		<img src={loaderImage} alt="loader" hidden={!props.isLoading} />
	</div>
);

SubmitButton.propTypes = {
  value: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default SubmitButton;
