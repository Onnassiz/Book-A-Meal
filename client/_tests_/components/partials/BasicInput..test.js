import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BasicInput, TextArea, Checkbox } from '../../../src/components/presentation/form/BasicInput';

const props = {
  name: 'Ben',
  label: 'Name',
  min: '100',
  type: 'text',
  value: 'Ben',
  onChange: jest.fn(),
  hasError: true,
  meal: { id: '123', name: 'Fried Rice' },
  isChecked: true,
};

const wrapper = shallow(<BasicInput {...props} />);
const textAreaWrapper = shallow(<TextArea {...props} />);

props.type = 'checkbox';

const checkBoxWrapper = shallow(<Checkbox {...props} />);

describe('<ShowErrors />', () => {
  it('Should render basic input component successfully', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should render textarea component successfully', () => {
    expect(toJson(textAreaWrapper)).toMatchSnapshot();
  });

  it('Should render checkbox component successfully', () => {
    expect(toJson(checkBoxWrapper)).toMatchSnapshot();
  });
});
