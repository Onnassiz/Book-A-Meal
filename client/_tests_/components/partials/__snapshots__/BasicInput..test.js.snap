// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowErrors /> Should render basic input component successfully 1`] = `
<div
  className="inputBox"
>
  <label
    htmlFor="Ben"
  >
    Name
  </label>
  <input
    autoComplete="Ben"
    className="has-error"
    id="Ben"
    min="100"
    name="Ben"
    onChange={[MockFunction]}
    type="text"
    value="Ben"
  />
</div>
`;

exports[`<ShowErrors /> Should render checkbox component successfully 1`] = `
<label
  className="checkbox"
>
   
  Fried Rice
   
  <span
    style={
      Object {
        "color": "#7a604a",
        "float": "right",
        "fontSize": 10,
      }
    }
  >
    Added by 
  </span>
  <input
    checked={true}
    onChange={[MockFunction]}
    type="checkbox"
    value="123"
  />
  <span
    className="check"
  />
</label>
`;

exports[`<ShowErrors /> Should render textarea component successfully 1`] = `
<div
  className="inputBox"
>
  <label
    htmlFor="Ben"
  >
    Name
  </label>
  <textarea
    className="has-error"
    cols="30"
    id="Ben"
    name="Ben"
    onChange={[MockFunction]}
    rows="6"
    value="Ben"
  />
</div>
`;
